dist/responsive_images.js

```diff
@@ -499,6 +510,90 @@ function getImgData( img ) {
     Object.values( imgData.intrinsicOrExtrinsicSizing ).includes( 'intrinsic' )
   );
 
+  // what was the impact of incorrect sizes values?
+
+  // helpers...
+  function geometricDistance(a, b) {
```

Comment:
In Chromium I used a [geometric mean](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/html/parser/html_srcset_parser.cc;l=396?q=geometric%20srcset%20f:.cc&ss=chromium) *between the candidates* as a tie breaker. I think this is slightly different

Reply:
Specific example where this would give you the wrong results:
1x and 2x candidates have a geometric mean of 1.44, so a 1.42 DPR screen would pick the 1x candidate, while this method would give you a ratio of 1.42 for the 1x and 1.408 for the 2x. I'm not sure if there are examples where the diff is more likely to be significant, but I wonder why you deviated from the geometric mean. Any particular reason?

Reply:
I twisted myself in a couple of knots until arriving on this solution, which was pleasingly functional and didn't require re-sorting, or special-casing for the first or last element. Glad it actually works! But I probably should have looked at the Chromium source and just copied that, first...